src/decorate.rb

```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.split.first(10)
```

Comment:
- [ ] you should return the corrected name back as a `String` currently since you are splitting the string it converts it to an `Array`. if you would like to do this you can append the `.join` method after so that it converts it back to a `String`